### ISSUE TITLE
ERROR [object Object] [object Object]

### ISSUE THREAD

**Issue by notrab** [GraphCMS/gatsby-source-graphcms#187]
Some users are reporting the following is returned when running Gatsby with the source plugin installed.

```bash
 ERROR 
[object Object] [object Object]

- gatsby-node.js:93 
```

We should improve error handling here.

**Comment by notrab** [comment]
Using `2.4.0` the message is now:

```bash
 ERROR 
gatsby-source-graphcms: Problem building GraphCMS nodes [object Object]
  Error: [object Object]
```

API endpoints in these examples are also set to public.

**Comment by notrab** [comment]
It appears after this PR https://github.com/GraphCMS/gatsby-source-graphcms/pull/166/files by @ynnoj was merged, that any API tokens that have just DRAFT access will throw the above error.

**Comment by killfill** [comment]
Maybe it would be a good idea to default to PUBLISHED **and** DRAFT in [here](https://github.com/GraphCMS/gatsby-source-graphcms/blob/main/gatsby-source-graphcms/src/gatsby-node.js#L62)? 

The failure seems to happen when the default PUBLISHED is used, and after you create a new graphcms project which only has DRAFT data. 😁

Btw, a better error output could be made [here](https://github.com/GraphCMS/gatsby-source-graphcms/blob/main/gatsby-source-graphcms/src/gatsby-node.js#L103) and [here](https://github.com/GraphCMS/gatsby-source-graphcms/blob/main/gatsby-source-graphcms/src/gatsby-node.js#L112), by doing something like

```javascript
new Error(JSON.stringify(response.errors))
```

Thanks!

**Comment by killfill** [comment]
Ok, its not 100% correct. Maybe the default should be DRAFT only, as having them both would require the access to be open for both env's at the endpoint side as well.

**Comment by notrab** [comment]
Yeah the reason it's set to `PUBLISHED` is due to that security around the data. `DRAFT` tends to be used for staging/dev mode.

@killfill I'd happily accept a PR for the `JSON.stringify` update. I think `response.errors` are thrown in a few places.

We really should get some contributor docs on here 😅

**Comment by irenama** [comment]
Hi @notrab are there any updates on this? 🙂

At the moment it seems like I can't query any `DRAFT` content from my GraphCMS project ...or is there a way around that?
All my 'mixed' (DRAFT & PUBLISHED) content tokens default to `PUBLISHED` items and the token with which I would only want to request `DRAFT` content breaks the build with the `[object Object] [object Object]` error.

I'm grateful for any insights, thanks! 🙏

**Comment by notrab** [comment]
@irenama I've asked @jpedroschmitz if he'll take a look at this when he has time. I haven't really spent a huge amount of time with this plugin (or Gatsby) to be very helpful - so I'll leave it to the expert @jpedroschmitz to investigate.

**Comment by jpedroschmitz** [comment]
Hey, I've been looking into this problem, and it [looks like there's a bug](https://github.com/gatsbyjs/gatsby/pull/28614/files#r542824222) on the Gatsby reporter.

I opened a PR (#188) that will display properly formatted messages, which allows us to understand the errors. In my case, while trying to reproduce it, I found that it was happening due to missing permissions on my GraphCMS token.
 
Let me know if there are any questions 💜

**Comment by jpedroschmitz** [comment]
We have released `2.5.2` on NPM that should fix the issue. 🎉 

To install, just run `yarn add gatsby-source-graphcms@latest` or `npm i gatsby-source-graphcms@latest`.

I'll be closing this issue for now. If there are any other problems you might have, don't hesitate to open an issue or reaching out. 👊🏻

**Comment by notrab** [comment]
I've added some [FAQs](https://github.com/GraphCMS/gatsby-source-graphcms/pull/191) to the README @irenama to help others seeing this issue.

I think the issue is due to the plugin having a default locale it uses to query, and the token not having the correct permissions. Unless both are configured the same, it's hard to solve, as far as I know.

If you have any ideas on how we can address this, I'd be happy to merge a PR.
